### PR TITLE
refactor: rewrite the sidebar, replace md components, optimize comple…

### PR DESCRIPTION
**Pull request by 2976151305**
## 🔨 Refactor
- rewrite the sidebar
- replace md components

## ⚡️ Performance
- optimize completion performance

## UI 🎨
- some mui to antd

### CODE REVIEW

File: components/chat/chat-container.tsx [hunk 1]
```diff
@@ -1,140 +1,127 @@
...
-};
+import MuiLoading from '../common/loading';
+import { Empty } from 'antd';
+import { useSearchParams } from 'next/navigation';
 
 const ChatContainer = () => {
+  const searchParams = useSearchParams();
+  const { scene, chatId, model, setModel } = useContext(ChatContext);
+  const chat = useChat({});
+  const initMessage = (searchParams && searchParams.get('initMessage')) ?? '';
```
Comment (Aralhi) [inline components/chat/chat-container.tsx:19]: `initMessage` in url is not a good ideal, maybe `PostMessage` or `Broadcast Channel`?

File: components/chat/chat-content.tsx [hunk 2]
```diff
@@ -1,70 +1,130 @@
...
   isChartChat?: boolean;
   onLinkClick: () => void;
 }
 
-const options: MarkdownToJSX.Options = {
-  overrides: {
-    code: ({ children, className }) => {
-      return (
-        <SyntaxHighlighter language="javascript" style={okaidia}>
+const mdComps: Parameters<typeof ReactMarkdown>['0']['components'] = {
```
Comment (Aralhi) [inline components/chat/chat-content.tsx:27]: `mdComps`, a full-name maybe better.

File: components/chat/chat-content.tsx [hunk 3]
```diff
@@ -1,70 +1,130 @@
...
...dv22uIEiLA4q7nvvCug8WTqzQveOH26fodo7g6uFe/a17W3+nFBAkRYENRdb1vkkz1CH9cPsVy/jrhr27PqMYvENYNlHAIesRiBYwRy0V+8iXP8+/fvX11Mr7L7ECueb/r48eMqm7FuI2BGWDEG8cm+7G3NEOfmdcTQw4h9/55lhm7DekRYKQPZF2ArbXTAyu4kDYB2YxUzwg0gi/41ztHnfQG26HbGel/crVrm7tNY+/1btkOEAZ2M05r4FB7r9GbAIdxaZYrHdOsgJ/wCEQY0J74TmOKnbxxT9n3FgGGWWsVdowHtjt9Nnvf7yQM2aZU/TIAIAxrw6dOnAWtZZcoEnBpNuTuObWMEiLAx1HY0ZQJEmHJ3HNvGCBBhY6jtaMoEiJB0Z29vL6ls58vxPcO8/zfrdo5qvKO+d3Fx8Wu8zf1dW4p/cPzLly/dtv9Ts/EbcvGAHhHyfBIhZ6NSiIBTo0LNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiECRCjUbEPNCRAhZ6NSiAARCjXbUHMCRMjZqBQiQIRCzTbUnAARcjYqhQgQoVCzDTUnQIScjUohAkQo1GxDzQkQIWejUogAEQo121BzAkTI2agUIkCEQs021JwAEXI2KoUIEKFQsw01J0CEnI1KIQJEKNRsQ80JECFno1KIABEKNdtQcwJEyNmoFCJAhELNNtScABFyNiqFCBChULMNNSdAhJyNSiEC/wGgKKC4YMA4TAAAAABJRU5ErkJggg=="
```
Comment (Aralhi) [inline components/chat/chat-content.tsx:109]: what this image is, why not use a image file.

File: components/chat/header/model-selector.tsx [hunk 4]
```diff
@@ -33,48 +33,30 @@ export function renderModelIcon(model?: string | undefined, props?: { width: num
...
-    return (
-      <>
-        {renderModelIcon(option.value)}
-        {option.label}
-      </>
-    );
-  }
   return (
-    <div
-      className={classNames({
```
Comment (Aralhi) [inline components/chat/header/model-selector.tsx:56]: set special width for `Select` is better.

File: components/chat/header/model-selector.tsx [hunk 5]
```diff
@@ -3,28 +3,28 @@
...
+import { ModelType } from '@/types/chart';
+import { Select } from 'antd';
 import { ICON_MAP } from '@/utils/constants';
 import classNames from 'classnames';
 import Image from 'next/image';
 import { useContext } from 'react';
 import { useTranslation } from 'react-i18next';
 
 interface Props {
-  size: 'sm' | 'md' | 'lg';
```
Comment (Aralhi) [inline components/chat/header/model-selector.tsx:14]: if you remove the `size` prop, then you should make sure that all references are properly styled.

File: hooks/use-chat.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          },
+          onclose() {
+            ctrl.abort();
+            onClose?.();
+          },
+          onerror(err) {
+            throw new Error(err);
+          },
+          onmessage: (event) => {
+            const message = event.data?.replaceAll('\\n', '\n');
```
[the remark this answers is not in the document]
Comment (2976151305) [reply hooks/use-chat.ts:65]: It is necessary to update here as well, as the backend response contains escape characters.

File: hooks/use-chat.ts [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  chatId: string;
+  context: string;
+  data?: Record<string, any>;
+  onMessage: (message: string) => void;
+  onClose?: () => void;
+  onDone?: () => void;
+  onError?: (content: string, error?: Error) => void;
+};
+
+const useChat = ({ queryAgentURL = '/api/v1/chat/completions', channel }: Props) => {
```
[the remark this answers is not in the document]
Comment (2976151305) [reply hooks/use-chat.ts:21]: done

File: components/chat/chat-content.tsx (hunk 2 above)
Comment (2976151305) [reply components/chat/chat-content.tsx:27]: rename to `markdownComponents`

File: components/chat/chat-content.tsx (hunk 3 above)
Comment (2976151305) [reply components/chat/chat-content.tsx:109]: download to local

**Review by Aralhi — APPROVED** [review]

**Pull request merged**